examples/html/supercells.html

```diff
@@ -0,0 +1,557 @@
+
+<script>
+"use strict"
+
+
+/*	----------------------------------
+	SUB/SUPERCELL CLASSES
+	----------------------------------
+	NOTE: these could be moved to src/cells and src/hamiltonian if used more often
+	Due to the point that these classes are very dependent on each other, they are currrently
+	kept separated to this example.
+*/
+
+/** A class encoding a host cell capable of holding @class SubCell cells
+ *  Intended to be used with SubCell and SubCellConstraint
+ */
+class SuperCell extends CPM.Cell {
+
+	/** The constructor of class SuperCell.
+	 * @param {object} conf - configuration settings of the simulation, containing the
+	 * relevant parameters. Note: this should include all constraint parameters.
+	 * @param {CellKind} kind - the cellkind of this cell, the parameters of kind are used 
+	 * when parameters are not explicitly overwritten
+	 * @param {CellId} id - the CellId of this cell (its key in the CPM.cells), unique identifier
+	 *  @param {CPMEvol} C - the CPM - used among others to draw random numbers
+	 * */
+	constructor (conf, kind, id, C) {
+		super(conf, kind, id, C)
+
+		/** the variable that is checked against subcells
+		 * @type{Number}
+		 */
+		this.host = this.id
+
+		/** holds the Supercell's @class{SubCell} SubCell objects 
+		 * @type{Array}
+		*/
+		this.subcells = []
+
+		/** encode cell-specific target volume for a simple growth algorithm such that the V can be halved after division
+		 * @type{Number}
+		*/
+		this.V = this.conf["V"][kind]
+	}
+
+
+	/**
+	 *  Halves target volume as post-division volume pressure can lead to artefacts
+	 * (with supercells specifically, the supercell can kill subcells through volume pressure if this is left out)
+	 * @param {Cell} parent - the parent (or other daughter) cell
+	 */ 
+	birth(parent){
+		super.birth(parent)
+		this.V /= 2
+		parent.V /=2
+	}
+
+	/**
+	 * Private function for SubCell, watch out when calling this. 
+	 * Adds a SubCell to the subcells array. 
+	 * @param {Cell} cell - the subcell to track from the supercell
+	 */ 
+	addSubCell(cell){
+		if (!this.subcells.includes(cell)){
+			this.subcells.push(cell)
+		} else {
+			throw("Tried to add subcell when already in subcells", this, cell)
+		}
+	}
+
+	/**
+	 * Private function for SubCell, watch out when calling this. 
+	 * Removes a SubCell to the subcells array. 
+	 * @param {Cell} cell - the subcell to stop tracking from the supercell
+	 */ 
+	removeSubCell(cell){
+		let ix =  this.subcells.indexOf(cell)
+		if (ix !==-1){
+			this.subcells.splice(ix, 1)
+		} else {
+			throw("Tried to remove subcell that was not tracked by this host", this, cell)
+		}
+	}
+
+	/**
+	 * Computes torus-enabled centroid for the pixels handed to it
+	 * Direct copy from Stat CentroidsWithTorusCorrection, but expects pixels instead of cellId
+	 * @param {Array} pixels for host and subcells
+	 * @return {ArrayCoordinate}  centroid of complete host+subcells space
+	 */ 
+	computeHostCentroid(pixels){
```

Comment:
The existing stat computes the pixels itself from an id, the pixels handed over here are a combination of multiple CellIds as this also contains subcells. I could also rewrite the stat and it would probably not be a big problem, but thought that this wouldn't be necessary. I can also just explain this reason in the comment above it.